File: 015-链表中倒数第k个结点/findkth.py
# -*- coding:utf-8 -*-
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def FindKthToTail(self, head, k):
        # write code here
        prior=head
        while k and prior is not None:
            prior=prior.next
            k-=1
        if k!=0:
            return None
        ans=head
        while prior:
            prior=prior.next
            ans=ans.next
        return ans

File: 015-链表中倒数第k个结点/findkth.cpp
class Solution {
public:
    ListNode* FindKthToTail(ListNode* pListHead, unsigned int k) {
        ListNode* prior=pListHead;
        while(k!=0 && prior!=nullptr){
            prior=prior->next;
            k--;
        }
        if (k>0){
            return nullptr;
        }
        ListNode* ans=pListHead;
        while(prior!=nullptr){
            ans=ans->next;
            prior=prior->next;
        }
        return ans;
    
    }
};
